packages/deployment/src/deployers/ProtocolDeployer.js

```diff
@@ -0,0 +1,298 @@
+  }
+
+  async loadGovernorIfNecessary() {
+    if (!this.config.governor) {
+      logger.warn('No governor specified, using default sender instead')
+      const sender = await this.environment.getSender()
+      this.config.governor = Governor(sender)
+    }
+  }
+
+  async loadOrDeployProtocol() {
+    const { protocol } = this.previousDeploy
+    const AragonProtocol = await this.environment.getArtifact('AragonProtocol', '@aragon/protocol-evm')
+
+    if (protocol && protocol.address) await this._loadAragonProtocol(AragonProtocol, protocol.address)
+    else await this._deployAragonProtocol(AragonProtocol)
+  }
+
+  async loadOrDeployDisputes() {
+    const { disputes } = this.previousDeploy
+    const DisputeManager = await this.environment.getArtifact('DisputeManager', '@aragon/protocol-evm')
+
+    if (disputes && disputes.address) await this._loadDisputes(DisputeManager, disputes.address)
+    else await this._deployDisputes(DisputeManager)
+  }
+
+  async loadOrDeployRegistry() {
+    const { registry } = this.previousDeploy
+    const GuardiansRegistry = await this.environment.getArtifact('GuardiansRegistry', '@aragon/protocol-evm')
+
+    if (registry && registry.address) await this._loadRegistry(GuardiansRegistry, registry.address)
+    else await this._deployRegistry(GuardiansRegistry)
+  }
+
+  async loadOrDeployVoting() {
+    const { voting } = this.previousDeploy
+    const Voting = await this.environment.getArtifact('CRVoting', '@aragon/protocol-evm')
+
+    if (voting && voting.address) await this._loadVoting(Voting, voting.address)
+    else await this._deployVoting(Voting)
+  }
+
+  async loadOrDeployTreasury() {
+    const { treasury } = this.previousDeploy
+    const Treasury = await this.environment.getArtifact('ProtocolTreasury', '@aragon/protocol-evm')
+
+    if (treasury && treasury.address) await this._loadTreasury(Treasury, treasury.address)
+    else await this._deployTreasury(Treasury)
+  }
+
+  async loadOrDeployPaymentsBook() {
+    const { paymentsBook } = this.previousDeploy
+    const PaymentsBook = await this.environment.getArtifact('PaymentsBook', '@aragon/protocol-evm')
+
+    if (paymentsBook && paymentsBook.address) await this._loadPaymentsBook(PaymentsBook, paymentsBook.address)
+    else await this._deployPaymentsBook(PaymentsBook)
+  }
+
+  async setModules() {
+    const sender = await this.environment.getSender()
+    const modulesGovernor = await this.protocol.getModulesGovernor()
+
+    if (modulesGovernor === sender) {
+      logger.info('Setting modules...')
+      const ids = [MODULES.DISPUTE_MANAGER, MODULES.TREASURY, MODULES.VOTING, MODULES.GUARDIANS_REGISTRY, MODULES.PAYMENTS_BOOK]
+      const implementations = [this.disputes, this.treasury, this.voting, this.registry, this.paymentsBook].map(i => i.address)
+      await this.protocol.setModules(ids, implementations)
+      logger.info('Caching modules...')
+      await this.protocol.cacheModules(implementations, ids)
```

Comment:
We've made a few changes to the module system, so perhaps this can be done in the same call now :)